src/rust/ide/view/src/documentation.rs

```diff
@@ -68,6 +78,7 @@ pub struct ViewModel {
     /// to EnsoGL shapes, and pass them to the DOM instead.
     overlay        : component::ShapeView<overlay::Shape>,
     display_object : display::object::Instance,
+    closures       : Rc<CloneCell<Vec<OptClosure>>>
```

Comment:
Why the `ViewModel` needs to be `CloneRef`? I did not have problems with that earlier, but now, if we clone ref it, we can easily leak accidenatally the closures, so I'm wondering why it needs to be clonable at all.

Reply:
Could you also explain e the live-time of the ViewModel? When it is created, who owns it and its copies, and when it will be deleted, so I could better understand how long the closures will live here? Also, add this info to the docs.

Reply:
Already explained by Adam.